### DISCUSSION TITLE
"You can't perform that action at this time" when I star a repository

### DISCUSSION THREAD

**Discussion by Helloyunho** [community/community#69366] [Repositories] [answered]
### Select Topic Area

Bug

### Body

Whenever I star a repository, I get this message at the very bottom of the page:
![CleanShot 2023-10-06 at 09 13 54@2x](https://github.com/community/community/assets/10192625/004cd5b8-5ed8-4fc2-aa30-baedb629edff)

The interesting part is that the feature actually works without issue(you can see the repository is successfully starred when you refresh the page), but just when you click the star button, that message shows up and the star button doesn't get updated(like it doesn't change from 'Star' -> 'Starred').

It's been about a week or two and it still is not resolved...

**Comment by MrDevel0per** [comment]
@Helloyunho I'm still running into the same issue in mid-October.

**Comment by Shailaputri** [comment]
Try clearing the web browser's cache and logging in again to the account. I was facing the same. Got resolved by this.

**Reply by Helloyunho** [reply]
I tried this just now but I'm still having the same issue

**Reply by mjfos2r** [reply]
I am also having the same issue and unfortunately that did not resolve it.

**Reply by MrDevel0per** [reply]
Thank you! Unfortunately, this didn't solve the issue for me.

**Reply by Usman-52** [reply]
I also having the same issue

**Comment by cesaryuan** [comment, marked as answer]
@Helloyunho @mjfos2r @MrDevel0per I also have this problem and have figured out why it happens. It's because Github uses the [Element.checkVisibility](https://chromestatus.com/feature/5163102852087808) API (at https://github.githubassets.com/assets/behaviors-934d11c22368.js), this API was added after Chrome 105.0, so any chrome version less than 105.0 will cause an error. Github has actually abandoned lower versions of Chrome.
## Solution 
To implement a polyfill for the `Element.checkVisibility` method. I provide a ready-made tampermonkey script to achieve this. [Fix "You can't perform that action at this time" for Github](https://greasyfork.org/zh-CN/scripts/478393-fix-you-can-t-perform-that-action-at-this-time-for-github)

**Update:** It seems that github just fix this issue last week which may need a few weeks to deploy to production. https://github.com/github/browser-support/pull/47

**Reply by Helloyunho** [reply]
This works! Hope GitHub deploys this soon to the public

**Reply by xiaohk** [reply]
Looking forward to the fix!

**Reply by landsman** [reply]
This still does not work on Safari. Can you please remove "Marked as answer"?

**Reply by Helloyunho** [reply]
I'm using Safari and it works
tho I'm using AdGuard to inject the script

**Comment by landsman** [comment]
I still seeing this issue in my latest Safari. What should I do?

**Reply by MrDevel0per** [reply]
I downloaded [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/), which [supports the API](https://developer.mozilla.org/en-US/docs/Web/API/Element/checkVisibility).

**Reply by landsman** [reply]
That's crazy. I operate on the latest browser and GitHub does not give a fuck.

**Reply by mjfos2r** [reply]
I also am still having this bug

**Comment by frankjonen** [comment]
Still doesn't work here. Just updated Safari as well.

**Comment by naveen-98** [comment]
Hey @Helloyunho  !

1. **Check Authentication:**
   Ensure that you are authenticated and logged into your GitHub account. If you are not logged in or there are authentication issues, GitHub may restrict certain actions.

2. **Verify Permissions:**
   Make sure that your GitHub account has the necessary permissions to star repositories. If your account has been restricted or if there are issues with permissions, it might result in the error.

3. **Review GitHub Status:**
   GitHub may experience service outages or disruptions. Check the [GitHub Status Page](https://www.githubstatus.com/) to see if there are any ongoing issues. If GitHub is experiencing problems, you may need to wait until the issues are resolved.

4. **Clear Browser Cache and Cookies:**
   Clear your browser's cache and cookies. Cached data may sometimes interfere with the proper functioning of GitHub features. After clearing the cache, try starring the repository again.

5. **Use a Different Browser or Device:**
   Try using a different web browser or device. Sometimes, browser extensions or configurations can cause issues. Using a different browser or device can help identify if the problem is browser-specific.

6. **Check for Browser Extensions:**
   Some browser extensions, especially those related to privacy or security, might interfere with website functionality. Try disabling browser extensions one by one to identify if any of them are causing the problem.

7. **GitHub Support:**
   If the issue persists, and you have ruled out common solutions, it may be worthwhile to reach out to [GitHub Support](https://support.github.com/contact) for assistance. Provide them with details about the issue and any error messages you encounter.

**Reply by Helloyunho** [reply]
As you can see in the answer, the problem is that GitHub uses the new `Element.checkVisibility` function that Safari doesn't seem to support yet. Using the polyfill in the answer fixes this problem.
PS: Did you even read the thread properly lol
PPS: When does GitHub update their polyfill? It's been a few months since this polyfill is merged...

**Reply by Usman-52** [reply]
i'm facing the same issue as you had (You can't perform that action at this time) Did you soloved the issue or still remain?

**Comment by ChristianVaughn** [comment]
Seems it still doesn't work on safari. surprised they are using a feature this new that all major browsers do not support yet.

**Reply by Helloyunho** [reply]
it's actually more like Safari not regularly applying their JS update from technical preview to official release

**Reply by xiaohk** [reply]
I agree with @ChristianVaughn. It feels GitHub has jumped the gun by using `checkVisibility` on production without a polyfill.

**Comment by bella8866** [comment]
Certain browser extensions, particularly those focused on privacy or security, could affect website functionality. Try disabling your extensions one by one to determine if any of them are causing the issue.
I had multiple extensions as I do work for my site [ihatecbts](https://ihatecbts.net/) that caused me this issue

**Comment by akaday** [comment]
i tried i pr i got the same error  in edge dev
